core/src/main/java/be/orbinson/aem/dictionarytranslator/models/Dictionary.java

```diff
@@ -18,4 +19,12 @@ public interface Dictionary {
     boolean isEditable();
 
     int getKeyCount();
+
+    /**
+     * Returns the dictionary resource having a lower or same ordinal, containing the given key and language and having at least one overlapping basename.
+     * @param key
+     * @param language
+     * @return the dictionary resource having a higher or same precedence if it exists, otherwise an empty Optional
+     */
+    Optional<Resource> getConflictingDictionary(String key, String language);
```

Comment:
this probably requires a third parameter `Collection<String> basenames` but I would rather implement this in context of #163.